tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ConfigFilesChecker.java

```diff
@@ -0,0 +1,399 @@
+    private Map<String, Object> getConfigEntry(File file) {
+        return (Map<String, Object>) new JsonSlurper().parse(file);
+    }
+
+    private boolean reflectConfigFileContainsErrors(File file) {
+        List<Map<String, Object>> entries = getConfigEntries(file);
+
+        if (entries.size() == 0) {
+            System.out.println("ERROR: empty reflect-config detected: " + file.toURI());
+            return true;
+        }
+
+        boolean containsErrors = containsDuplicatedEntries(entries, file);
+        for (var entry : entries) {
+            containsErrors |= checkTypeReachable(entry, file);
+            containsErrors |= containsEntriesNotFromLibrary(entry, file);
+        }
+
+        return containsErrors;
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean resourceConfigFileContainsErrors(File file) {
+        Map<String, Object> entries = getConfigEntry(file);
+        List<Map<String, Object>> bundles = (List<Map<String, Object>>) entries.get("bundles");
+        Map<String, Object> resources = (Map<String, Object>) entries.get("resources");
+
+        boolean containsErrors = false;
+        if (resources != null) {
+            List<Map<String, Object>> includes = (List<Map<String, Object>>) resources.get("includes");
+            List<Map<String, Object>> excludes = (List<Map<String, Object>>) resources.get("excludes");
+
+            if (listNullOrEmpty(includes) && listNullOrEmpty(excludes) && listNullOrEmpty(bundles)){
+                System.out.println("ERROR: empty resource-config detected: " + file.toURI());
+                return true;
+            }
+
+            // check include entries
+            if (includes != null) {
+                containsErrors |= containsDuplicatedEntries(includes, file);
+
+                for (var entry : includes) {
+                    containsErrors |= checkTypeReachable(entry, file);
+                }
+            }
+
+            // check exclude entries
+            if (excludes != null) {
+                containsErrors |= containsDuplicatedEntries(excludes, file);
+
+                for (var entry : excludes) {
+                    containsErrors |= checkTypeReachable(entry, file);
+                    containsErrors |= containsEntriesNotFromLibrary(entry, file);
+                }
+            }
+        }
+
+        return containsErrors;
+    }
+
+    private boolean checkOldSerializationConfig(File file) {
+        List<Map<String, Object>> entries = getConfigEntries(file);
+        
+        if (entries.isEmpty()) {
+            System.out.println("ERROR: empty serialization-config detected: " + file.toURI());
+            return true;
+        }
+
+        boolean containsErrors = containsDuplicatedEntries(entries, file);
+        for (var entry : entries) {
+            containsErrors |= checkTypeReachable(entry, file);
+            containsErrors |= containsEntriesNotFromLibrary(entry, file);
+        }
```

Comment:
This block could be a separate function which we then use here and in the two places where we check the new serialization config